radicle-surf/src/vcs/git/repo.rs

```diff
@@ -142,69 +190,69 @@ impl<'a> RepositoryRef<'a> {
         Ok(namespaces?.into_iter().collect())
     }
 
-    pub(super) fn ref_history<R>(&self, reference: R) -> Result<History, Error>
-    where
-        R: Into<Ref>,
-    {
-        let reference = match self.which_namespace()? {
-            None => reference.into(),
-            Some(namespace) => reference.into().namespaced(namespace),
-        }
-        .find_ref(self)?;
-        self.to_history(&reference)
-    }
-
     /// Get the [`Diff`] between two commits.
-    pub fn diff(&self, from: &Rev, to: &Rev) -> Result<Diff, Error> {
-        let from_commit = self.rev_to_commit(from)?;
-        let to_commit = self.rev_to_commit(to)?;
+    pub fn diff<R: FromRevision>(&self, from: R, to: R) -> Result<Diff, Error> {
```

Comment:
We're turning these into `Commit`s so this should use the `IntoCommit` trait :) The phrase "statement of intent" comes to mind. Since revisions can be used to refer to any objects they're too general, but `IntoCommit` tells us that we're definitely going through `Commit` objects.
```suggestion
    pub fn diff<R: IntoCommit>(&self, from: R, to: R) -> Result<Diff, Error> {
```

Reply:
Ah that's a fair point! I missed this function [`diff_blobs`](https://docs.rs/git2/0.15.0/git2/struct.Repository.html#method.diff_blobs).

> Are we sure this diff will only support commits going forward? (I know it's hard to predict the future, but I was thinking maybe it's beneficial to use `FromRevision` here).

I'm happy to go forward with `FromRevision`! Good catch :)